packages/@aws-cdk/core-synthesizer/lib/staging-stack.ts

```diff
@@ -0,0 +1,61 @@
+import * as ecr from '@aws-cdk/aws-ecr';
+import * as s3 from '@aws-cdk/aws-s3';
+import { DockerImageAssetSource, RemovalPolicy, Stack, StackProps } from '@aws-cdk/core';
+import { Construct, IConstruct } from 'constructs';
+
+/**
+ * Information on how a Staging Stack should look.
+ */
+export interface IStagingStack extends IConstruct {
+  /**
+   * // TODO
+   */
+  readonly stagingBucket: s3.Bucket;
+
+  /**
+   * // TODO
+   */
+  readonly stagingRepos: Record<string, ecr.Repository>;
+
+  /**
+   * // TODO
+   */
+  locateRepo(asset: DockerImageAssetSource): ecr.Repository;
+}
+
+/**
+ * Staging Stack Properties
+ */
+export interface StagingStackProps extends StackProps {
+}
+
+/**
+ * A default Staging Stack
+ */
+export class StagingStack extends Stack implements IStagingStack {
+  /**
+   * // TODO
+   */
+  public readonly stagingBucket: s3.Bucket;
+
+  /**
+   * // TODO
+   */
+  public readonly stagingRepos: Record<string, ecr.Repository>;
+
+  constructor(scope: Construct, id: string, props: StagingStackProps = {}) {
+    super(scope, id, props);
+
+    this.stagingBucket = new s3.Bucket(this, 'StagingBucket', {
+      autoDeleteObjects: true,
+      removalPolicy: RemovalPolicy.DESTROY,
+    });
+
+    this.stagingRepos = {};
+  }
+
+  public locateRepo(asset: DockerImageAssetSource): ecr.Repository {
+    // TODO: not the source hash! construct path
```

Comment:
Well yeah and create on demand, rite? :)

Reply:
did that, but i actually can't find the construct path... `DockerImageAssetSource` is an interface, not a construct.